fs/xfstests.py

```diff
@@ -221,6 +229,72 @@ def setUp(self):
+                build.make(e2fsprogs_build_dir)
+                build.make(e2fsprogs_build_dir, extra_args='install')
+
+            if self.fs_to_test == "xfs":
+                if self.detected_distro.name in ['centos', 'fedora', 'rhel']:
+                    libini_path = process.run("ldconfig -p | grep libini",
+                                              verbose=True, ignore_status=True)
+                    if not libini_path:
+                        # Build libini.h as it is needed for xfsprogs
+                        libini_dir = os.path.join(self.teststmpdir, 'libini')
+                        if not os.path.exists(libini_dir):
+                            os.makedirs(libini_dir)
+                        git.get_repo('https://github.com/benhoyt/inih',
+                                     destination_dir=libini_dir)
+                        os.chdir(libini_dir)
+                        process.run("meson build", verbose=True)
+                        libini_build_dir = os.path.join(libini_dir, 'build')
+                        if os.path.exists(libini_build_dir):
+                            os.chdir(libini_build_dir)
+                            process.run("meson install", verbose=True)
+                        else:
+                            self.fail('Something went wrong while building \
+                                      libini. Please check the logs.')
+                # Build xfs progs
+                xfsprogs_dir = os.path.join(self.teststmpdir, 'xfsprogs')
+                if not os.path.exists(xfsprogs_dir):
+                    os.makedirs(xfsprogs_dir)
+                xfsprogs_url = self.params.get('xfsprogs_url')
+                git.get_repo(xfsprogs_url, destination_dir=xfsprogs_dir)
+                os.chdir(xfsprogs_dir)
+                build.make(xfsprogs_dir)
+                build.make(xfsprogs_dir, extra_args='install')
+
+            if self.fs_to_test == "btrfs":
+                # Build btrfs progs
+                btrfsprogs_dir = os.path.join(self.teststmpdir, 'btrfsprogs')
+                if not os.path.exists(btrfsprogs_dir):
+                    os.makedirs(btrfsprogs_dir)
+                btrfsprogs_url = self.params.get('btrfsprogs_url')
+                git.get_repo(btrfsprogs_url, destination_dir=btrfsprogs_dir)
+                os.chdir(btrfsprogs_dir)
+                process.run("./autogen.sh", verbose=True)
+                process.run("./configure --prefix=/usr/local/ --disable-documentation", verbose=True)
+                build.make(btrfsprogs_dir)
+                build.make(btrfsprogs_dir, extra_args='install')
+
+        # Check versions of fsprogs
+        fsprogs_ver = process.system_output("mkfs.%s -V" % self.fs_to_test,
+                                            ignore_status=True,
+                                            shell=True).decode("utf-8")
+        self.log.info(fsprogs_ver)
 
         if process.system('which mkfs.%s' % self.fs_to_test,
```

Comment:
Hey Sachin, all the distros I've tested seem to have `/usr/local/bin` before `/usr/sbin` in their $PATH. In my tests, the script is correctly able to pick up the updated mkfs.btrfs from `/usr/local/bin`. Can you please provide the avocado run logs and $PATH value of the system you observe this behavior on. I can take a look.

Reply:
> Hey Sachin, all the distros I've tested seem to have `/usr/local/bin` before `/usr/sbin` in their $PATH. In my tests, the script is correctly able to pick up the updated mkfs.btrfs from `/usr/local/bin`. Can you please provide the avocado run logs and $PATH value of the system you observe this behavior on. I can take a look.

On SLES it's the other way around.
 
~ echo $PATH
/sbin:/usr/sbin:/usr/local/sbin:/root/bin:/usr/local/bin:/usr/bin:/bin
~

Reply:
Just to summarize the discussing we had on Slack regarding this:

Basically, there are times when the xfstests also call mkfs  and when it does so, they use the one in `$PATH` and I couldn't find a way to change this. So I think we will have to make sure the binaries are installed at right path instead of explicitly using `/usr/local/bin/mkfs`

So we agreed upon something like the follows:
```
if SLES:
    --prefix=/sbin
else RHEL:
    --prefix=/usr/local
```

I've updated this PR to reflect this change. The behavior is now something as follows:
- Install libs and other stuff in `/usr/local` for both RHEL and SuSE
- Install the binaries (mkfs etc) in `/usr/local/{bin,sbin}` for RHEL and `/sbin` for SuSE